### PR TITLE
DUPLO-26393 TF:GKE: Getting error while creating exclusions for resource 'duplocloud_gcp_infra_maintenance_window'

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview

Fixed exclusion issue and added delete context and changed attribute properties based on it
## Summary of changes
Fixed exclusion issue and added delete context and changed attribute properties based on it

This PR does the following:

- Change daily_maintenance_time format validation
- Using post api for delete context
- Changed properties for recurring_window attributes
- updated documentation

## Testing performed

- [ ] Using unit tests
- [ ] Manually, on my local system
- [✔︎ ] Manually, on a remote test system

## Describe any breaking changes

- ...

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-26393 TF:GKE: Getting error while creating exclusions for resource 'duplocloud_gcp_infra_maintenance_window'](https://app.clickup.com/t/8655600/DUPLO-26393)

File: duplocloud/resource_duplo_gcp_infra_maintenance_window.go [hunk 1]
```diff
@@ -114,7 +126,7 @@ func resourceInfrastructureMaintenanceWindowRead(ctx context.Context, d *schema.
 	token := d.Id()
 	infraName := strings.Split(token, "/")[1]
 	log.Printf("[TRACE] resourceInfrastructureMaintenanceWindowRead(%s): start", infraName)
-
+	time.Sleep(10 * time.Second) //for sync with gcp
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_gcp_infra_maintenance_window.go:129]: Can we use some API for wait.

Comment (Duplo-nikv) [reply duplocloud/resource_duplo_gcp_infra_maintenance_window.go]: added possible retry on the getapi. Only efficient logic I can think of

File: duplocloud/resource_duplo_gcp_infra_maintenance_window.go [hunk 2]
```diff
@@ -155,7 +178,19 @@ func resourceInfrastructureMaintenanceWindowCreateOrUpdate(ctx context.Context,
...
 
 func resourceInfrastructureMaintenanceWindowDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	return nil
+	infraName := d.Get("infra_name").(string)
+
+	c := m.(*duplosdk.Client)
+	rq := duplosdk.DuploGcpInfraMaintenanceWindow{}
+	err := c.CreateGCPInfraMaintenanceWindow(infraName, &rq)
+	if err != nil {
+		return diag.Errorf("resourceInfrastructureMaintenanceWindowCreateOrUpdate cannot create maintenance window for infra %s error: %s", infraName, err.Error())
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_gcp_infra_maintenance_window.go:187]: Change the error message

**Review by tahir-duplo — APPROVED** [review]
LGTM

**Pull request merged**